GDAL/OGR read via pyogrio on Windows is failing

From the Python Wheels build:

```
2025-11-24T03:24:58.1938825Z E       sedonadb._lib.SedonaError: DataSourceError: '/C:/Users/RUNNER~1/AppData/Local/Temp/tmp17ns9xkz/temp.fgb' does not exist in the file system, and is not recognized as a supported dataset name.
2025-11-24T03:24:58.1939457Z 
2025-11-24T03:24:58.1939611Z ..\venv-test\lib\site-packages\sedonadb\context.py:217: SedonaError
2025-11-24T03:24:58.1939958Z =========================== short test summary info ===========================
2025-11-24T03:24:58.1940989Z FAILED D:\a\sedona-db\sedona-db\python\sedonadb\tests\test_datasource.py::test_read_ogr_projection - sedonadb._lib.SedonaError: DataSourceError: '/C:/Users/RUNNER~1/AppData/Local/Temp/tmpzn4_47de/temp.fgb' does not exist in the file system, and is not recognized as a supported dataset name.
2025-11-24T03:24:58.1942829Z FAILED D:\a\sedona-db\sedona-db\python\sedonadb\tests\test_datasource.py::test_read_ogr_multi_file - sedonadb._lib.SedonaError: DataSourceError: '/C:/Users/RUNNER~1/AppData/Local/Temp/tmpp4uaafo7/partition=part_%7Bc%7D/ciVEqw5kWOckkExp.fgb' does not exist in the file system, and is not recognized as a supported dataset name.
2025-11-24T03:24:58.1944637Z FAILED D:\a\sedona-db\sedona-db\python\sedonadb\tests\test_datasource.py::test_read_ogr_filter - sedonadb._lib.SedonaError: DataSourceError: '/C:/Users/RUNNER~1/AppData/Local/Temp/tmp17ns9xkz/temp.fgb' does not exist in the file system, and is not recognized as a supported dataset name.
2025-11-24T03:24:58.1945677Z ========== 3 failed, 1112 passed, 1097 skipped, 2 xfailed in 10.26s ===========
2025-11-24T03:25:01.6602704Z ##[error]cibuildwheel: Command pytest D:\a\sedona-db\sedona-db\python\sedonadb/tests -vv failed with code 1. 
```

I think this is because the Objstore -> File path conversion doesn't currently work for driver letters.

--- a/ci/scripts/wheels-build-windows.ps1
+++ b/ci/scripts/wheels-build-windows.ps1
@@ -73,6 +73,8 @@ $env:PATH += ";$scriptDirectory\windows"
 $env:GEOS_LIB_DIR = "$vcpkgLibDirectory"
 $env:GEOS_VERSION = "3.13.0"
 
+$env:MATURIN_PEP517_ARGS='--features s2geography,pyo3/extension-module'
+
 # Some CMake configurations needs this separately from the toolchain file
 $env:CMAKE_PREFIX_PATH="$vcpkgInstalledDirectory"
 $env:OPENSSL_ROOT_DIR="$vcpkgInstalledDirectory"

